unit-tests/live/syncer/test-throughput.cpp

```diff
@@ -0,0 +1,248 @@
+                    return;
+                auto frame_num = f.get_frame_number();
+                auto actual_fps = (float)f.get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_FPS);
+                auto frame_arrival = f.get_frame_metadata(RS2_FRAME_METADATA_FRAME_TIMESTAMP); // usec
+                if (stream_type == "Infrared 1")
+                {
+                    _ir_frames_arrival_info.push_back({ (float)frame_arrival, actual_fps });
+                    if (actual_fps != prev_fps) // exposure new value is finally set
+                    {
+                        auto t_now = std::chrono::system_clock::now();
+                        _delta = (float)std::chrono::duration_cast<std::chrono::seconds>(t_now - t_start).count();
+                    }
+                    prev_fps = actual_fps;
+                }
+                if (stream_type == "Color")
+                    _rgb_frames_arrival_info.push_back({ (float)frame_arrival, actual_fps });
+            };
+
+            float delta = (float)std::chrono::duration_cast<std::chrono::seconds>(t_end - t_start).count();
+            while (delta < RECEIVE_FRAMES_TIME)
+            {
+                if (delta > RECEIVE_FRAMES_TIME/2)
+                    set_ir_exposure(); // set exposure to 18000 after 5 sec
+
+                auto fs = _sync.wait_for_frames();
+                for (const rs2::frame& ff : fs)
+                {
+                    process_frame(ff);
+                }
+                t_end = std::chrono::system_clock::now();
+                delta = (float)std::chrono::duration_cast<std::chrono::seconds>(t_end - t_start).count();
+            }
+        }  
+        void analysis(configuration test, std::vector <std::pair<float, float>>& frames_arrival_info, std::string stream_type = "Infrared 1")
+        {
+            if (frames_arrival_info.empty())
+                return;
+            float fps = 0.0f;
+            while (fps < 10) // skip unstable frames at the beginning
+            {
+                fps = frames_arrival_info.front().second;
+                frames_arrival_info.erase(frames_arrival_info.begin());
+            }
+            std::vector<std::pair<float, float>>  vec[2];
+            for (auto& frm : frames_arrival_info)
+            { 
+                auto fps_ratio = frm.second / fps; // used ratio instead of precised fps because syncer doesn't change fps to exact value when running RGB
+                if (fps_ratio > 0.9)
+                    vec[0].push_back(frm);
+                else
+                    vec[1].push_back(frm);
+            }
+
+            float delta[2] = { _delta, RECEIVE_FRAMES_TIME - _delta };
+            int i = 0;
+            for (auto& v : vec)
+            {
+                if (v.empty())
+                    continue;
+                auto actual_fps = v.front().second;
+                float calc_fps = (float)v.size() / delta[i];
```

Comment:
The timing does not account for the first ten frames dropped for each actual FPS.
Check the time of arrival for the first and  the last frame to establish it.